app/services/wallets/create_interval_wallet_transactions_service.rb

```diff
@@ -15,7 +15,10 @@ def call
             source: :interval,
             invoice_requires_successful_payment: recurring_transaction_rule.invoice_requires_successful_payment?
           }
-        )
+        }
+        job_params[:params][:metadata] = recurring_transaction_rule.metadata unless recurring_transaction_rule.metadata.empty?
+
+        WalletTransactions::CreateJob.perform_later(job_params)
```

Comment:
why not? is the key not included when value is empty to avoid updating the tests?

```
  WalletTransactions::CreateJob.perform_later(
          organization_id: wallet.organization.id,
          params: {
            wallet_id: wallet.id,
            paid_credits: paid_credits(recurring_transaction_rule),
            granted_credits: granted_credits(recurring_transaction_rule),
            source: :interval,
            invoice_requires_successful_payment: recurring_transaction_rule.invoice_requires_successful_payment?,
            metadata: recurring_transaction_rule.metadata
          }
  )
```

is there an issue if `metadata` key is included with an empty hash as value?

Reply:
no problem